### PR TITLE
Add steps to funnel trends

### PR DESCRIPTION
**Pull request by rcmarron**
## Changes

Adds support for `funnel_step_to` and `funnel_step_from` to the front-end of funnel trends.

In doing so, this fixes a bug where the funnel trends graph often showed the trend of users who went from step 0 -> step 1 even if there are more steps in the funnel. (second bullet of #8179 ). This was happening because:
* The back-end supported `funnel_step_X` properties, but the front-end didn't
* A bug caused the front-end to default these properties being set as 0 and 1 for most funnels (also fixed)

## How did you test this code?

Verified:
* Changing the selector updates the funnel trend graph as expected
* When building a funnel, if the step picker hasn't been explicitly set, it stays set as the max range of the funnel as steps new are added on both funnel trends and time to convert.

### CODE REVIEW

File: frontend/src/scenes/funnels/funnelUtils.ts [hunk 1]
```diff
@@ -348,16 +348,23 @@ export const getClampedStepRangeFilter = ({
...
+            ...(stepRange as FunnelStepRangeEntityFilter),
+            funnel_from_step,
+            funnel_to_step: clamp(
+                stepRange?.funnel_to_step ?? filters.funnel_to_step ?? maxStepIndex,
+                funnel_from_step + 1,
+                maxStepIndex
+            ),
+        }
     }
+    return {}
```
Comment (alexkim205) [inline frontend/src/scenes/funnels/funnelUtils.ts:367]: I think differentiating between having the steps set and unset will cause some funky behavior here. To the user there is no difference when a funnel step is set to 0 vs when it's set to 0 by default.

**Review by alexkim205 — DISMISSED** [review]

**Comment by alexkim205** [comment]
Nice! Can't believe this bug sat in prod for so long. Some thoughts (feel free to ignore):

* I wonder if its worth it to strip out `funnel_from` and `funnel_to` in funnel steps views. The backend ignores them anyway for that display type and clutters the shape of the api params.


Bugs:

* Adding steps in the funnel view doesn't seem to update steps picker as well. Only happens when first step is explicitly selected as "Step 1" in dropdown.

https://user-images.githubusercontent.com/13460330/151236765-32108d5e-de52-452c-90d1-6a5be1d468fd.mov

File: frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelExclusionsFilter.tsx [hunk 2]
```diff
@@ -31,8 +31,8 @@ function ExclusionRowSuffix({
     }
 
     const onChange = (
-        funnel_from_step: number = stepRange.funnel_from_step,
-        funnel_to_step: number = stepRange.funnel_to_step
+        funnel_from_step: number | undefined = stepRange.funnel_from_step,
```
[the remark this answers is not in the document]
Comment (rcmarron) [reply frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelExclusionsFilter.tsx:34]: Yeah, looks like you can't use the `?` notiation when you also have an initializer for the parameter - hence the `| undefined`

File: frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelStepsPicker.tsx [hunk 3]
```diff
@@ -44,27 +44,25 @@ export function FunnelStepsPicker(): JSX.Element | null {
         <Row className="funnel-options-inputs">
             <span className="text-muted-alt">between</span>
```
[the remark this answers is not in the document]
Comment (rcmarron) [reply frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelStepsPicker.tsx:45]: Good call

File: frontend/src/scenes/funnels/funnelUtils.ts (hunk 1 above)
Comment (rcmarron) [reply frontend/src/scenes/funnels/funnelUtils.ts]: Good call on this. Fixed it as part of the bug below!

**Comment by rcmarron** [comment]
Just fixed the bug + made the suggested changes. Also added some tests to cover the bug.

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**